README.md

```diff
@@ -81,99 +94,111 @@ Try the workshop at https://apprunnerworkshop.com
-
-Options specific to the Stack component.
-
-```ts
-interface StackOptions
+interface AppResourceOptions
 ```
 
 #### `remapCloudControlResource`
 
-This optional method can be implemented to define a mapping to override and/or provide an implementation for a CloudFormation resource type that is not (yet) implemented in the AWS Cloud Control API (and thus not yet available in the Pulumi AWS Native provider). Pulumi code can override this method to provide a custom mapping of CloudFormation elements and their properties into Pulumi CustomResources, commonly by using the AWS Classic provider to implement the missing resource.
+This optional method can be implemented to define a mapping to override and/or
+provide an implementation for a CloudFormation resource type that is not (yet)
+implemented in the AWS Cloud Control API (and thus not yet available in the
+Pulumi AWS Native provider). Pulumi code can override this method to provide a
+custom mapping of CloudFormation elements and their properties into Pulumi
+CustomResources, commonly by using the AWS Classic provider to implement the
+missing resource.
 
 ```ts
-remapCloudControlResource(element: CfnElement, logicalId: string, typeName: string, props: any, options: pulumi.ResourceOptions): { [key: string]: pulumi.CustomResource } | undefined
+remapCloudControlResource(logicalId: string, typeName: string, props: any, options: pulumi.ResourceOptions): { [key: string]: pulumi.CustomResource } | undefined
 ```
 
 Parameters:
-* `element`: The full CloudFormation element object being mapped.
 * `logicalId`: The logical ID of the resource being mapped.
 * `typeName`: The CloudFormation type name of the resource being mapped.
 * `props`: The bag of input properties to the CloudFormation resource being mapped.
 * `options`: The set of Pulumi ResourceOptions to apply to the resource being mapped.
 
-Returns an object containing one or more logical IDs mapped to Pulumi resources that must be created to implement the mapped CloudFormation resource, or else undefined if no mapping is implemented.
+Returns an object containing one or more logical IDs mapped to Pulumi resources
+that must be created to implement the mapped CloudFormation resource, or else
+undefined if no mapping is implemented.
+
+#### `appId`
+
+This is a unique identifier for the application. It will be used in the names of the
+staging resources created for the application. This `appId` should be unique across apps.
+
+### `Stack`
+
+A Construct that represents an AWS CDK stack deployed with Pulumi. In order to
+deploy a CDK stack with Pulumi, it must derive from this class.
 
-#### `create`
+#### `constructor`
 
 Create and register an AWS CDK stack deployed with Pulumi.
 
 ```ts
-create(name: string, ctor: typeof Stack, opts?: pulumi.CustomResourceOptions): StackComponent
+constructor(app: App, name: string, options?: StackOptions)
 ```
 
 Parameters:
+* `app`: The Pulumi CDK App
 * `name`: The _unique_ name of the resource.
```

Comment:
Isn't this rather the name of the construct because we're in CDK land?

Reply:
Actually no, this is a special Pulumi CDK Stack which requires the app to be the special Pulumi App that is passed to the callback function.

Reply:
Ah ok, got it!